project/src/main/puzzle/combo-counters.gd

```diff
@@ -0,0 +1,70 @@
+extends Node2D
+"""
+Combo indicators which appear when the player clears a line in puzzle mode.
+"""
+
+export (NodePath) var piece_manager_path: NodePath
+export (NodePath) var playfield_path: NodePath
+
+# Counter which alternates between -1, 0 and 1 to vary the x position of consecutive line clears
+var _position_counter := 0
```

Comment:
This seems overengineered and doesn't serve the intended purpose.

If two consecutive line clears have the same position, let's just randomly shift it right or left one.

Reply:
Redesigned to track the previous X coordinate, and shift the combo counters around if the algorithm picks the same X coordinate twice in a row.